gitpod-io/gitpod
# Add repo-level editor restriction

## Description

- [x] Need to be rebased after https://github.com/gitpod-io/gitpod/pull/19542

## Related Issue(s)

Fixes EXP-1484, EXP-1262, EXP-1264

## How to test

- Try to update both repo and org-level available editors
- Test on create workspace page if the option and error message is expected

Examples to load create workspace pages:
- Change contextUrl on browser manually
- Add search param `editor` on url manually
- Update selected repo
- Update selected editor then update selected repo
- ...

## Documentation


#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - hw-repo-editor</li>
	<li><b>🔗 URL</b> - <a href="https://hw-repo-editor.preview.gitpod-dev.com/workspaces" target="_blank">hw-repo-editor.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - hw-repo-editor-gha.23804</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-hw-repo-editor%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-core-dev" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
</details>

/hold
